test/org/cyverse/datacite_test.clj

```diff
@@ -0,0 +1,196 @@
+    (test-missing-fields "datacite.publicationyear")
+    (test-blank-fields "datacite.publicationyear")))
+
+(deftest test-missing-resource-type
+  (testing "DataCite file generation with missing resource type."
+    (test-missing-fields "datacite.resourcetype")
+    (test-blank-fields "datacite.resourcetype")))
+
+(deftest test-all-missing-fields
+  (testing "DataCite file generateion with all required fields missing."
+    (apply test-missing-fields (keys min-attrs))
+    (apply test-blank-fields (keys min-attrs))))
+
+(deftest test-creator-with-name-id
+  (testing "DataCite file with creator name identifier."
+    (test-datacite "datacite/creator-name-id.xml" (assoc min-attrs "creatorNameIdentifier" "foo"))))
+
+(deftest test-multiple-creators-without-name-id
+  (testing "DataCite file with multiple creators without the name identifier."
+    (->> [["datacite.creator" "Somebody Else"]
+          ["creatorAffiliation" "University of Somewhere"]]
+         (concat min-attrs)
+         (test-datacite "datacite/multiple-creators.xml"))))
+
+(deftest test-group-with-different-numbers-of-required-attr-values
+  (testing "DataCite file for metadata group with different numbers of required attribute values"
+    (->> [["datacite.creator" "Somebody Else"]]
+         (concat min-attrs)
+         (test-exception #"^These attributes must have the same number of values"))))
+
+(deftest test-group-with-more-optional-attr-values-than-required-attr-values
+  (testing "DataCite file for metadata group with more optional attribute values than required attribute values."
+    (->> [["creatorNameIdentifier" "foo"]
+          ["creatorNameIdentifier" "bar"]]
+         (concat min-attrs)
+         (test-exception #"^None of the attributes, (?:[^,]+, )+may have more values than any of the attributes"))))
+
+(deftest test-subject
+  (testing "DataCite file generation with subjects."
+    (->> [["Subject" "foo,bar,baz"]
+          ["Subject" "quux"]
+          ["Subject" "blrfl"]]
+         (concat min-attrs)
+         (test-datacite "datacite/subjects.xml"))))
+
+(deftest test-contributors
+  (testing "DataCite file generation with contributors."
+    (->> [["contributorName" "Somebody Else"]
+          ["contributorType" "DataCollector"]
+          ["contributorName" "Nobody Else"]
+          ["contributorType" "Editor"]]
+         (concat min-attrs)
+         (test-datacite "datacite/contributors.xml"))))
+
+(deftest test-alternate-identifiers
+  (testing "DataCite file generation with alternate identifiers."
+    (->> [["AlternateIdentifier" "the-alternate-id"]
+          ["alternateIdentifierType" "ARK"]
+          ["AlternateIdentifier" "the-other-alternate-id"]
+          ["alternateIdentifierType" "FOOID"]]
+         (concat min-attrs)
+         (test-datacite "datacite/alternate-ids.xml"))))
+
+(deftest test-empty-optional-repeating-field
+  (testing "DataCite file generation with alternate identifiers."
```

Comment:
Should this be more specific to this test?

Reply:
Ah, yes, I forgot to update the description. thanks for catching that.